File: src-tauri/src/commands/external/seventv.rs
use crate::dto::external::emote::EmoteEntry;
use crate::dto::external::seventv::ChannelResult;
use crate::services;

#[tauri::command]
pub async fn seventv_get_global_emotes() -> Result<Vec<EmoteEntry>, String> {
    services::external::seventv::get_global_emotes().await
}

#[tauri::command]
pub async fn seventv_get_channel_emotes(channel_id: String) -> Result<ChannelResult, String> {
    services::external::seventv::get_channel_emotes(channel_id).await
}

File: src/lib/api/external/seventv.ts
import type { EmoteEntry } from "../../types/external/emote.ts";
import type { ChannelResult } from "../../types/external/seventv.ts";
import { invokeCommand, type InvokeOptions } from "../utils.ts";

export function seventvGetGlobalEmotes(
  options?: InvokeOptions,
): Promise<EmoteEntry[]> {
  return invokeCommand("seventv_get_global_emotes", undefined, options);
}

export type SeventvGetChannelEmotesParams = {
  channelId: string;
};

export function seventvGetChannelEmotes(
  params: SeventvGetChannelEmotesParams,
  options?: InvokeOptions,
): Promise<ChannelResult> {
  return invokeCommand("seventv_get_channel_emotes", params, options);
}
